certificateshop-logic/src/main/java/com/epam/esm/dao/impl/jpa/JpaOrderDaoImpl.java

```diff
@@ -0,0 +1,220 @@
+                .executeUpdate();
+    }
+
+    /**
+     * Finds all {@link Order} entity in the database.
+     *
+     * @return List of the {@link Order} objects.
+     */
+    @Override
+    public List<Order> findAll() {
+        Query query = entityManager.createNativeQuery(FIND_ALL_ENTITIES_SQL);
+        List<Object[]> resultList = query.getResultList();
+        List<List<Object>> result = convertListOfArrayToListOfLists(resultList);
+        return getEntities(result);
+    }
+
+    /**
+     * Finds {@link Optional <Order>} in the database by the id of the {@link Order}.
+     *
+     * @param id is the {@link long} to find.
+     * @return {@link Optional<Order>}.
+     */
+    @Override
+    public Optional<Order> findById(long id) {
+        Query query = entityManager.createNativeQuery(FIND_ENTITY_BY_ID_SQL).setParameter(1, id);
+        List<Object[]> resultList = query.getResultList();
+        List<List<Object>> result = convertListOfArrayToListOfLists(resultList);
+        List<Order> orders = getEntities(result);
+        return orders.stream().findFirst();
+    }
+
+    /**
+     * Updates the {@link Order}.
+     *
+     * @param order is the value of the {@link Order} to update.
+     */
+    @Override
+    public void update(Order order) {
+
+    }
+
+    /**
+     * Removes the {@link Order} object from the database.
+     *
+     * @param id is the value of the {@link long} to find.
+     */
+    @Override
+    public void delete(long id) {
+
+    }
+
+    /**
+     * Finds {@link Optional<Order>} in the database by the id of the {@link Order}.
+     *
+     * @param name is the {@link String} to find.
+     * @return {@link Optional<Order>}.
+     */
+    @Override
+    public Optional<Order> findByName(String name) {
+        Query query = entityManager.createNativeQuery(FIND_ENTITY_BY_NAME_SQL).setParameter(1, name);
+        List<Object[]> resultList = query.getResultList();
+        List<List<Object>> result = convertListOfArrayToListOfLists(resultList);
+        List<Order> users = getEntities(result);
+        if (users.isEmpty()) {
+            return Optional.empty();
+        }
+        return users.stream().findFirst();
+    }
+
+    private List<List<Object>> convertListOfArrayToListOfLists(List<Object[]> resultList) {
+        List<List<Object>> result = new ArrayList();
+        for (Object[] objects : resultList) {
+            List<Object> listOfObjects = new ArrayList<>(Arrays.asList(objects));
+            result.add(listOfObjects);
+        }
+        return result;
+    }
+
+    private List<Order> getEntities(List<List<Object>> result) {
+        ResultSet resultSet;
+        List<Order> orders = new ArrayList<>();
+        try {
+            resultSet = listToSetConverter.getResultSet(ORDER_HEADERS, result);
+            orders = orderExtractor.extractData(resultSet);
+        } catch (Exception exception) {
+            throw new RuntimeException(exception);
+        }
```

Comment:
Why do you use default exceptions instead of your own?

Reply:
Because I need just throw the same type of exception and i am know what can cause it (and i am sure that it won't be thrown). That is why it doesn't matter for the functionality how this exception will be named. Am i mistaking?